src/core/coap/coap.cpp

```diff
@@ -160,13 +160,13 @@ Message *CoapBase::InitMessage(Message *aMessage, Type aType, Uri aUri)
     return aMessage;
 }
 
-Message *CoapBase::InitResponse(Message *aMessage, const Message &aResponse)
+Message *CoapBase::InitResponse(Message *aMessage, const Message &aRequest)
```

Comment:
FYI - on pretty check failure:

```
/home/runner/work/openthread/openthread/src/core/coap/coap.hpp:890:14: error: function 'ot::Coap::CoapBase::InitResponse' has a definition with different parameter names [readability-inconsistent-declaration-parameter-name,-warnings-as-errors]
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
    Message *InitResponse(Message *aMessage, const Message &aResponse);
1 warning treated as error
             ^
/home/runner/work/openthread/openthread/src/core/coap/coap.cpp:163:20: note: the definition seen here
Message *CoapBase::InitResponse(Message *aMessage, const Message &aRequest)
                   ^
/home/runner/work/openthread/openthread/src/core/coap/coap.hpp:890:14: note: differing parameters are named here: ('aResponse'), in definition: ('aRequest')
    Message *InitResponse(Message *aMessage, const Message &aResponse);
             ^                                              ~~~~~~~~~
                                                            aRequest
```

Reply:
Thanks, I now fixed the definition. As a side question: I saw that aRequest / aResponse when supplied in the ot... CoAP APIs are not checked for nullptr - they will just be used. Is that intentional?

Reply:
> I saw that aRequest / aResponse when supplied in the ot... CoAP APIs are not checked for nullptr - they will just be used. Is that intentional?

Yes, it's intentional. The general rule for public/platform OpenThread C APIs is that pointer input variables shouldn't be `nullptr` and if a null value is allowed, it should be explicitly mentioned in the API documentation. There should be no run-time check (and returning error) for this so to avoid the code size overhead caused by excessive nullptr checks.

Instead, there's a configuration option, `OPENTHREAD_CONFIG_ASSERT_CHECK_API_POINTER_PARAM_FOR_NULL`, which adds `assert` checks to ensure input pointer arguments aren't null. These assertions are often hidden within [`AsCoreType<>()`](https://github.com/openthread/openthread/blob/93e838102c05af3cb34584033c1999c805b97120/src/core/common/as_core_type.hpp#L56-L88) functions or they are direct calls to `AssertPointerIsNotNull`. 

While this is the general approach, and almost all OT APIs follow it, there may be some inconsistencies in some older/earlier OT APIs (e.g., still doing null check and returning error, and/or not doing assert check) :)